interface/modules/custom_modules/oe-module-weno/src/Services/WenoValidate.php

```diff
@@ -0,0 +1,264 @@
+<?php
+
+namespace OpenEMR\Modules\WenoModule\Services;
+
+use GuzzleHttp\Client;
+use GuzzleHttp\Exception\GuzzleException;
+
+class WenoValidate extends ModuleService
+{
+    private string $requestUrl = 'https://online.wenoexchange.com/webapi/restapi/WenoManage';
+    private string $messageID;
+    private string $userEmail;
+    private string $md5UserPassword;
+    private mixed $encryptionKey;
+    private Client $client;
+
+    public function __construct()
+    {
+        parent::__construct();
+        $this->generateMessageID();
+        $this->setMessageProperties();
+        $this->client = new Client();
+    }
+
+    /**
+     * @return void
+     */
+    private function generateMessageID(): void
+    {
+        $random_string = substr(str_shuffle("0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"), 0, 5);
```

Comment:
if you need to generate a cryptographically secure ramdom value, then would do (with `use OpenEMR\Common\Utils\RandomGenUtils`):
```
$random_string = RandomGenUtils::produceRandomString(5, '0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ');
```